Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCAutoFollow.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NPCAutoFollow : DialogueAutoplay
+{
+    private bool isFollowing = false;
+    protected GameObject targetObject;
+
+    protected NavMeshAgent agent;
+
+
+    void Start()
+    {
+        agent = Utils.GetRequiredComponent<NavMeshAgent>(this);
+    }
+
+
+    void Update()
+    {
+        if(isFollowing)
+        {
+            // TODO: only chase character within a certain area
+            ChaseTarget();
+        }
+    }
+
+    // Temporary way to trigger interaction (Remove once Trigger Interaction is integrated)
+    // Remove this and box collider on NPC when done
+    private void OnTriggerEnter(Collider other)
+    {
+        if(other.gameObject.layer == LayerMask.NameToLayer(Constants.LAYER_PLAYER))
+        {
+            TriggerAutoplay();
+
+            targetObject = other.gameObject;
+            TriggerFollow();
+        }
+    }
+
+    public new void TriggerInteraction(GameObject target)
+    {
+        base.TriggerInteraction(target);
+
+        targetObject = target;
+        TriggerFollow();
+    }
+
+
+    void TriggerFollow()
+    {
+        if(targetObject == null)
+        {
+            Debug.LogError("Target to follow must be assigned");
+            return;
+        }
+
+        if(!isFollowing)
+        {
+            isFollowing = true;
+        }
+    }
+
+
+    void StopFollow()
+    {
+        if(isFollowing)
+        {
+            isFollowing = false;
+        }
+    }

```

Comment:
This can probably just be 
`isFollowing = false;`, cause the outcome is the same anyways.

Reply:
Or the function can just be removed, unless you're planning on adding more functionality to the function.

Reply:
Should I keep it if I plan on calling it somewhere else again or if that's the only functionality then should I just do `isFollowing = false;` everytime I need the character to stop following.

Reply:
If you plan on calling it elsewhere in this file or even later making it public or something, then I'd definitely keep the function!